app/models/detector.rb

```diff
@@ -1,5 +1,15 @@
 # frozen_string_literal: true
 
+# Detectors are a generic representation of specific tests performed by the
```

Comment:
I think `specific tests` as a phrase may lead to confusion when working in a context in which we have test suites that mean something else. Maybe `specific checks` or similar?